python/dgl/contrib/dis_kvstore.py

```diff
@@ -374,26 +340,62 @@ def start(self):
                         id=None,
                         data=None,
                         c_ptr=None)
-                    for i in range(self._client_count):
-                        _send_kv_msg(self._sender, back_msg, i)
-                    self._barrier_count = 0
-            # FINAL message
+                    for client_id in range(self._client_count):
+                        _send_kv_msg(self._sender, back_msg, client_id)
+                    self._barrier_count = 0  
+            # Final message              
             elif msg.type == KVMsgType.FINAL:
-                print("Exit KVStore service, server ID: %d" % self._server_id)
+                print("Exit KVStore service %d" % self.get_id())
                 break # exit loop
             else:
                 raise RuntimeError('Unknown type of kvstore message: %d' % msg.type.value)
 
+            # garbage collection
             self._garbage_msg.append(msg)
-            if len(self._garbage_msg) > 1000:
+            if len(self._garbage_msg) > GARBAGE_COLLECTION_COUNT:
                 _clear_kv_msg(self._garbage_msg)
```

Comment:
Why we use a list to store bag of msgs? Can we call _clear_ky_msg each time? The is huge performance overhead?

Reply:
I'm not sure which way is good. I just think that if we invoke _clear_kv_msg every time, the latency will be higher.